packages/sync/src/links/compositeQueueLink.ts

```diff
@@ -1,11 +1,23 @@
 import { OfflineQueueLink, TYPE_MUTATION } from "./OfflineQueueLink";
 import { LocalDirectiveFilterLink } from "./LocalDirectiveFilterLink";
 import { DataSyncConfig } from "../config";
-import { ApolloLink, concat } from "apollo-link";
+import { ApolloLink, Operation } from "apollo-link";
+import { onError } from "apollo-link-error";
+import { RetryLink } from "apollo-link-retry";
 
 export const compositeQueueLink = (config: DataSyncConfig, filter?: TYPE_MUTATION): ApolloLink => {
   const offlineLink = new OfflineQueueLink(config, filter);
   offlineLink.openQueueOnNetworkStateUpdates();
+  const errorLink = onError(({ networkError, operation }) => {
+    if (networkError) {
+      operation.variables.__networkError = networkError;
+    }
+  });
+  const retryLink = new RetryLink({
```

Comment:
This should be done in separate file with description what corner case this is solving